adal/src/androidTest/java/com/microsoft/aad/adal/OauthTests.java

```diff
@@ -162,215 +173,198 @@ public void testParseIdTokenNegativeInvalidEncodedTokens() {
                 "loginhint 1234567890-+=;'", null, null, null);
-        Object oauth = createOAuthInstance(request);
-        Method m = ReflectionUtils.getTestMethod(oauth, "getCodeRequestUrl");
-
-        String actual = (String) m.invoke(oauth);
-        assertTrue(
-                "Matching message",
-                actual.contains("http://www.something.com/oauth2/authorize?response_type=code&client_id=client+1234567890-%2B%3D%3B%21%23%24+++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&resource=resource%2520urn%3A%21%23%24++++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&redirect_uri=redirect+1234567890&state="));
+        final Oauth2 oauth2 = createOAuthInstance(request);
+        final String actualCodeRequestUrl = oauth2.getCodeRequestUrl();
+        assertTrue("Matching message",
+                actualCodeRequestUrl.contains("http://www.something.com/oauth2/authorize?response_type=code&client_id=client+1234567890-%2B%3D%3B%21%23%24+++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&resource=resource%2520urn%3A%21%23%24++++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&redirect_uri=redirect+1234567890&state="));
         assertTrue("Matching loginhint",
-                actual.contains("login_hint=loginhint+1234567890-%2B%3D%3B%27"));
+                actualCodeRequestUrl.contains("login_hint=loginhint+1234567890-%2B%3D%3B%27"));
 
         // without login hint
-        Object requestWithoutLogin = createAuthenticationRequest("http://www.something.com",
+        final AuthenticationRequest requestWithoutLogin = createAuthenticationRequest(
+                "http://www.something.com",
                 "resource%20urn:!#$    &'( )*+,/:  ;=?@[]",
                 "client 1234567890-+=;!#$   &'( )*+,/:  ;=?@[]", "redirect 1234567890", "", null,
                 null, null);
 
-        Object oauthWithoutLoginHint = createOAuthInstance(requestWithoutLogin);
+        final Oauth2 oauthWithoutLoginHint = createOAuthInstance(requestWithoutLogin);
 
-        actual = (String) m.invoke(oauthWithoutLoginHint);
+        final String actualCodeRequestNoLoginhint = oauthWithoutLoginHint.getCodeRequestUrl();
         assertTrue(
                 "Matching message",
-                actual.contains("http://www.something.com/oauth2/authorize?response_type=code&client_id=client+1234567890-%2B%3D%3B%21%23%24+++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&resource=resource%2520urn%3A%21%23%24++++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&redirect_uri=redirect+1234567890&state="));
-        assertFalse("Without loginhint", actual.contains("login_hint=loginhintForCode"));
+                actualCodeRequestNoLoginhint.contains(
+                        "http://www.something.com/oauth2/authorize?response_type=code&client_id=client+1234567890-%2B%3D%3B%21%23%24+++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&resource=resource%2520urn%3A%21%23%24++++%26%27%28+%29*%2B%2C%2F%3A++%3B%3D%3F%40%5B%5D&redirect_uri=redirect+1234567890&state="));
```

Comment:
I should move up the Matching message, and move the contains also one line up. Will update